### DISCUSSION TITLE
RetroPie Install

### DISCUSSION THREAD

**Discussion by tgaldi** [diasurgical/DevilutionX#1145] [Q&A] [answered]
Hey - I just setup a Pi4 with RetroPie and trying to get Devilution installed. When I launch it in the ports menu I get the launch menu and then it pauses and returns to the ports menu.

I grabbed the latest release devilutionx-linux-armhf.7z.

Made sure sdl dependencies are installed.

I've created the folder retropie\roms\ports\diablo:
![image](https://user-images.githubusercontent.com/17083706/110537407-a2fe1900-80f0-11eb-8177-1bc56e266b9e.png)

and I put this script (diablo.sh) in the ports directory: 
`#!/bin/bash`
`"/opt/retropie/supplementary/runcommand/runcommand.sh" 0 _PORT_ "diablo" ""`

I set up the config file:
![image](https://user-images.githubusercontent.com/17083706/110537491-bf9a5100-80f0-11eb-9c8b-dbd8e6b79182.png)
`diablo = "/home/pi/RetroPie/roms/ports/diablo/devilutionx --data-dir /home/pi/RetroPie/roms/ports/diablo"`
`default = "diablo"`

**Comment by AJenbo** [comment]
Feel free to add this to the wiki rather than the discussion section, unless you are having problems with it?
https://github.com/diasurgical/devilutionX/wiki

**Reply by tgaldi** [reply]
I haven't been able to get the game to launch.

**Comment by AJenbo** [comment, marked as answer]
Ok, you are not really describing anything specifically related to DevilutionX but just retro pie settings, so unless you have an error message or something I think you need to contact the Retro Pi community. You could add try doing this with a simple test application to narrow dow the issue you are having.

**Reply by tgaldi** [reply]
Ah, solved it. Needed to set executable permissions:  chmod u+x devilutionx

**Reply by tgaldi** [reply]
Is the a list/guide for .ini setting?

**Reply by AJenbo** [reply]
I'm not sure what you are asking for, you're not the only one asking, but once people figure it out they don't bother to write one either.

I think the names are fairly self-explanatory, but you can also view the options comment from the code here if that helps you.
https://github.com/diasurgical/devilutionX/blob/master/Source/diablo.h#L22

**Reply by tgaldi** [reply]
Thanks, I was just going through the source looking for this.

I'm not sure if I'm using it incorrectly, but the settings are not working. I've put a diablo.ini inside the folder, and as a test tried to set the sound volume to 0:

![image](https://user-images.githubusercontent.com/17083706/110553406-ecf1f980-8106-11eb-8c03-46ae5e8f1a58.png)

`[DevilutionX]`
`SoundVolume=0`

**Reply by AJenbo** [reply]
It generates the file itself, if you don't want to use the normal path given by the os then overwrite it with `--save-dir` or `--config-dir`

You got the name and probably value wrong, but use the one it generates instead of trying to fill it out beforehand.

**Reply by tgaldi** [reply]
Excellent. --config-dir didn't work but --save-dir works.

Changed it in the emulators.cfg:
`"/home/pi/RetroPie/roms/ports/diablo/devilutionx --save-dir /home/pi/RetroPie/roms/ports/diablo"`

**Reply by tgaldi** [reply]
There wouldn't happen to be a way to reduce the UI for more gameplay real-estate?

**Comment by AJenbo** [comment]
you can increase the game area, that would make it take less of the total

**Reply by tgaldi** [reply]
ok and last question I swear - the sounds are sometimes distorted, is there a way to fix this?

Oh and how do I contribute all this information to the wiki?

**Reply by AJenbo** [reply]
> ok and last question I swear

couldn't even stick to it for one post :D

1: you probably have a buggy version of SDL2 on ARM, you need to update it, check your distribution for options

2: add a new page to the wiki
https://github.com/diasurgical/devilutionX/wiki/_new

**Reply by tgaldi** [reply]
Says I don't have permission to update this wiki.

**Reply by AJenbo** [reply]
see if you can edit now
